packages/client-utils/test/smart-wallet-with-sequence.test.ts

```diff
@@ -0,0 +1,188 @@
+    sequence: number;
+  }> = [];
+
+  private networkSequence: () => number;
+
+  private shouldSimulateSequenceConflicts = false;
+
+  constructor(getNetworkSequence: () => number) {
+    this.networkSequence = getNetworkSequence;
+  }
+
+  enableSequenceConflictSimulation() {
+    this.shouldSimulateSequenceConflicts = true;
+  }
+
+  async sendBridgeAction(
+    action: any,
+    fee: any,
+    memo: any,
+    signerData: any,
+  ): Promise<SubmitTxResponse> {
+    // Simulate network delay
+    await new Promise(resolve => setTimeout(resolve, 10));
+
+    const currentNetworkSequence = this.networkSequence();
+
+    // Simulate sequence mismatch if enabled and sequence is out of sync
+    if (
+      this.shouldSimulateSequenceConflicts &&
+      signerData.sequence < currentNetworkSequence
+    ) {
+      throw new Error(
+        `Broadcasting transaction failed with code 32 (codespace: sdk). Log: account sequence mismatch, expected ${currentNetworkSequence}, got ${signerData.sequence}: incorrect account sequence`,
+      );
+    }
+
+    // Record successful transaction
+    const method = action.method || 'sendBridgeAction';
+    this.submittedTransactions.push({
+      method,
+      sequence: signerData.sequence,
+    });
+
+    return {
+      code: 0,
+      height: 3321450 + this.submittedTransactions.length,
+      transactionHash: `hash_${method}_${signerData.sequence}`,
+      sequence: signerData.sequence,
+    };
+  }
+
+  getSubmittedTransactions() {
+    return this.submittedTransactions;
+  }
+}
+
+test('handles concurrent offers and actions with correct sequence management', async t => {
+  const mockFetch = createMockFetchAccountInfo();
+  const mockWallet = new MockSigningSmartWalletKit(() =>
+    mockFetch.getNetworkSequence(),
+  );
+
+  const logs: string[] = [];
+  const log = (...args: any[]) => logs.push(args.join(' '));
+
+  const sequenceManager = await makeSequenceManager(
+    { log },
+    { address: 'agoric1test', fetchAccountInfo: mockFetch.fetch },
+  );
+
+  const walletWithSequence = makeSmartWalletWithSequence(
+    {
+      signingSmartWalletKit: mockWallet as any as SigningSmartWalletKit,
+      sequenceManager: sequenceManager as any,
+      log,
+    },
+    { chainId: 'agoricdev-25' },
+  );
+
+  const promises = [
+    walletWithSequence.executeOffer({ id: 'offer1' } as any),
+    walletWithSequence.executeOffer({ id: 'offer2' } as any),
+    walletWithSequence.sendBridgeAction({ method: 'test' } as any),
+    walletWithSequence.executeOffer({ id: 'offer3' } as any),
+  ];
+
+  const results = (await Promise.all(promises)) as any;
+
+  t.is(results.length, 4);
+  results.forEach((result: any) => {
+    t.is(result.code, 0);
+    t.truthy(result.transactionHash);
+  });
+
+  const submittedTransactions = mockWallet.getSubmittedTransactions();
+  t.is(submittedTransactions.length, 4);
+
+  const sequences = submittedTransactions.map(tx => tx.sequence);
+  t.deepEqual(sequences, [100, 101, 102, 103]);
```

Comment:
See [packages/internal/tools/ava-assertions.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/internal/tools/ava-assertions.js):
```suggestion
  arrayIsLike(t, results, [
    // prettier-ignore
    { code: 0, transactionHash: 'hash_sendBridgeAction_100', height: 3321451, sequence: 100 },
    // prettier-ignore
    { code: 0, transactionHash: 'hash_sendBridgeAction_101', height: 3321452, sequence: 101 },
    // prettier-ignore
    { code: 0, transactionHash: 'hash_test_102', height: 3321453, sequence: 102 },
    // prettier-ignore
    { code: 0, transactionHash: 'hash_sendBridgeAction_103', height: 3321454, sequence: 103 },
  ]);
  
  const submittedTransactions = mockWallet.getSubmittedTransactions();
  arrayIsLike(t, submittedTransactions, [
    { method: 'sendBridgeAction', sequence: 100 },
    { method: 'sendBridgeAction', sequence: 101 },
    { method: 'test', sequence: 102 },
    { method: 'sendBridgeAction', sequence: 103 },
  ]);
```